src/translator/definitions.h

```diff
@@ -21,6 +21,24 @@ template <class T, typename... Args> UPtr<T> UNew(Args &&... args) {
 
 template <class T> UPtr<T> UNew(UPtr<T> p) { return UPtr<T>(p); }
 
+class MemoryGift{
```

Comment:
I'm sorry to change my mind here, but the memory management is looking really questionable here.  It seems MemoryGift should own the memory.  

Moreover, to support multiple platforms with alignment there's actually different free.  

Might I suggest https://github.com/kpu/intgemm/blob/master/intgemm/aligned.h but one can get rid of the template argument if it's always storing void.  In theory you have this in the submodule.  Could copy here though.